instrumentation/kafka-clients/src/main/java/brave/kafka/clients/KafkaTracing.java

```diff
@@ -72,6 +84,21 @@ public Builder remoteServiceName(String remoteServiceName) {
       return this;
     }
 
+    /**
+     * When no trace context or extra fields have been added to an incoming consumed message propagation,
+     * records from the same partition are been traced under the same parent id, sharing the same trace.
+     *
+     * <b/>If true, this will share a trace for messages received as part of
+     * a batch with no trace context, and create an individual trace by message; leading to 1 trace
+     * per poll operation.
+     *
+     * @since 5.10
+     */
+    public Builder shareTraceOnConsumption(boolean shareTraceOnConsumption) {
```

Comment:
I think it'd be nice to indicate that this applies to new traces in the name itself. `shareNewTraceOnConsumption`? 

Though for me it's easier to grok with the opposite - `splitNewTraceOnConsumption`. What do you think?

Reply:
Also I know the class is called consumer but I think receive might be easier to understand.

Reply:
Thanks! I'm not convinced about naming either hehe. I'll rephrase my explanation to have a better background for naming.

Reply:
tend to agree, since `shareTraceOnConsumption` is the current behaviour it might stand out more if the flag refers to enabling the opposite behaviour . `splitTraceOnConsumption` , `spanPerMessage`, `noSpanPerBatch`, 

Above names are still slightly misleading as they could make ppl (like me) think that it allows you to break a span, whereas the flag only applies when no parent trace is around. `spanPerMessageWhenNoParentTraceFound` ?

Reply:
would `shareRootSpanOnReceive` or `rootSpanOnReceiveBatch` be a better name? (I think `span...WhenNoParentTrace` ~= `rootSpan`)

Reply:
btw @jorgheymans I think we should have a similar flag to do not continue a trace on the consumer side (e.g. instead that removing headers on the producer), that would be related with this one. for instance: if producer is instrumented and injecting trace context, on the consumer side we can decide to do not continue the same trace context, and just create a new root span. Then, this flag can help to opt-out of sharing that root span between consumed messages in a batch.

Will tackle this on an additional PR as it will impact other messaging instrumentation.

Reply:
@jeqo so maybe this should be bundled together as one feature then, where the `separate-span-if-no-existing-root` and `force-start-new-trace` are just different strategies of this feature ? This 'something' would allow one to relatively easily define new processing like 'if the message header contains value X we want to start a new trace, otherwise join existing'. Or is that a bridge too far ?

Reply:
I'd say that (1) starting a new trace on consume is independent from (2) reusing/sharing rootSpan when consuming a batch. Where (1) is shared across messaging instrumentations, probably should be added at that level of abstraction; and (2) is specific to Kafka so far as we don't have other consumer instrumentation polling batches, but could eventually be moved to `messaging` as well. So, I'd take these as 2 diff features.

About `if (header=x) then start new trace`, I haven't thought about this as initial use-case. My first answer will be to do this once you have consumed a message, as part of your code. But we can discuss this further here https://github.com/openzipkin/brave/issues/1039